### DISCUSSION TITLE
Allow arrow keys only in insert mode?

### DISCUSSION THREAD

**Discussion by agoodshort** [m4xshen/hardtime.nvim#68] [Q&A] [answered]
I am trying to allow the arrow keys only if in insert mode. But whatever I do, I keep being blocked, how would you achieve that? Here is [my hardtime config](https://github.com/agoodshort/nvim/blob/7bd3696650120dcc53d54a161eb83e36576bc113/lua/agoodshort/plugins/hardtime.lua) for reference.

Also, could you please clarify what `""` and `"i"` does in:
```lua
   disabled_keys = {
      ["<Up>"] = { "", "i" },
      ["<Down>"] = { "", "i" },
      ["<Left>"] = { "", "i" },
      ["<Right>"] = { "", "i" },
   },
```

Thank you very much for this great plugin, I'm using it since your first post on Reddit and it really removed a lot of bad habits.

**Comment by m4xshen** [comment, marked as answer]
First of all thanks for the feedback!

Here's the help files to help you understand `""` and `"i"`:

`:h nvim_set_keymap()`
```
      • {mode}  Mode short-name (map command prefix: "n", "i", "v", "x", …) or
                "!" for |:map!|, or empty string for |:map|.
```

`:h map-table`
```
         Mode  | Norm | Ins | Cmd | Vis | Sel | Opr | Term | Lang |  
Command        +------+-----+-----+-----+-----+-----+------+------+  
[nore]map      | yes  |  -  |  -  | yes | yes | yes |  -   |  -   |
n[nore]map     | yes  |  -  |  -  |  -  |  -  |  -  |  -   |  -   |
[nore]map!     |  -   | yes | yes |  -  |  -  |  -  |  -   |  -   |
i[nore]map     |  -   | yes |  -  |  -  |  -  |  -  |  -   |  -   |
c[nore]map     |  -   |  -  | yes |  -  |  -  |  -  |  -   |  -   |
v[nore]map     |  -   |  -  |  -  | yes | yes |  -  |  -   |  -   |
x[nore]map     |  -   |  -  |  -  | yes |  -  |  -  |  -   |  -   |
s[nore]map     |  -   |  -  |  -  |  -  | yes |  -  |  -   |  -   |
o[nore]map     |  -   |  -  |  -  |  -  |  -  | yes |  -   |  -   |
t[nore]map     |  -   |  -  |  -  |  -  |  -  |  -  | yes  |  -   |
l[nore]map     |  -   | yes | yes |  -  |  -  |  -  |  -   | yes  |
```

In the default config `""` and `i` stands for the `map` and `imap`, which includes Norm, Vis, Sel, Opr and Ins. Therefore the arrow keys are disabled inside all modes except Cmd, Term and Lang mode. I allow the Cmd and Term modes because the arrow key is necessary, and for the Lang mode it is less used so I just ignore it.

That being said, If you only want to allow the key in insert mode, you can just set it to `{ "" }` (or `{ "", "c", "t", "l" }` if you really means insert mode ONLY).

**Reply by agoodshort** [reply]
Thanks a lot for you quick and really detailed answer! I kinda knew what the `"i"` was but wasn't really sure where `""` was pointing to.

Back to my main question/issue, I actually found out where my issue was. I used `<DOWN>` instead of `<Down>`... Can't believe it took me so long to find out. [Here is the related issue/comment](https://github.com/m4xshen/hardtime.nvim/issues/29#issuecomment-1683688849) to my struggle :smile: 

Thanks again!
